src/lesson4/Calculator.java

```diff
@@ -0,0 +1,180 @@
+        return currentValue;
+    }
+
+    public double division(double num1, double num2) {
+        currentValue = num1 / num2;
+        return currentValue;
+    }
+
+    public void commandC() {
+        currentValue = 0;
+        calculatorLook(currentValue);
+    }
+
+    public void commandBackspace() { //only for currentValue
+        double number = currentValue;
+        String numberStr = Double.toString(number);
+        int i = 0;
+        while (numberStr.toCharArray()[numberStr.length() - 1] == '.' | numberStr.toCharArray()[numberStr.length() - 1] == '0') {
+            i++;
+            String numberStrEdited = numberStr.substring(0, numberStr.length() - i);
+            if ((numberStrEdited.toCharArray()[numberStrEdited.length() - 1] != ('.')) & (numberStrEdited.toCharArray()[numberStrEdited.length() - 1] != ('0'))) {
+                if (numberStrEdited.length() == 1) {
+                    commandC();
+                    break;
+                }
+                numberStrEdited = numberStrEdited.substring(0, numberStr.length() - (i + 1));
+                currentValue = Double.parseDouble(numberStrEdited);
+                calculatorLook(currentValue);
+                break;
+            }
+        }
+        if (!(numberStr.toCharArray()[numberStr.length() - 1] == '.' | numberStr.toCharArray()[numberStr.length() - 1] == '0')) {
+            String numberStrEdited = numberStr.substring(0, numberStr.length() - (i + 1));
+            currentValue = Double.parseDouble(numberStrEdited);
+            calculatorLook(currentValue);
+        }
+    }
+
+
+    public boolean stringIsDouble(String string) {
+        char[] stringInCharArray = string.toCharArray();
+        boolean isDouble = false;
+        int countOfComas = 0;
+        if (string.length() > 12) {
+            System.err.println("Too many characters in your double value!");
+            return false;
+        }
+        for (int i = 0; i < string.length(); i++) {
+            if (stringInCharArray[i] == ('1')
+                    | (stringInCharArray[i] == ('2')
+                    | stringInCharArray[i] == ('3')
+                    | stringInCharArray[i] == ('4')
+                    | stringInCharArray[i] == ('5')
+                    | stringInCharArray[i] == ('6')
+                    | stringInCharArray[i] == ('7')
+                    | stringInCharArray[i] == ('8')
+                    | stringInCharArray[i] == ('9')
+                    | stringInCharArray[i] == ('0'))) isDouble = true;
+            else if (stringInCharArray[i] == ('.')) {
+                if (countOfComas < 1) {
+                    countOfComas++;
+                } else System.err.println("Too many comas in your double value!");
+            }
+            else return false;
+        }
+        return isDouble;
+    }
+
+    public boolean isCommand(String command) {
+        return switch (command) {
+            case ("+"), ("-"), ("*"), ("/") -> true;
```

Comment:
looks bad, probably  creation array of all command will be good idea, you can iterate through it and return true if string is in the array and false if not, which  is bed without break at the end